components/style/properties/properties.mako.rs

```diff
@@ -2516,8 +2522,17 @@ impl<'a> StyleBuilder<'a> {
         flags: ComputedValueFlags,
         visited_style: Option<Arc<ComputedValues>>,
     ) -> Self {
+        debug_assert!(parent_style.is_some() == parent_style_ignoring_first_line.is_some());
+        // FIXME(bz): I wish I could assert that either
+        // parent_style == parent_style_ignoring_first_line or parent_style is
+        // a ::first-line style... But ComputedValues doesn't know its pseudo.
```

Comment:
We _could_ track it. The Servo one doesn't because layout doesn't use it, but there's no reason you can't poke at the `ServoStyleContext`s `mPseudo` from here actually... You could add  a `cfg`'d `is_first_line_style` method on `ComputedValues`, if you think it's worth the effort.